src/strings.js

```diff
@@ -40,54 +40,112 @@ function rewritePatternForDotAll(pattern) {
   return cachedRegExp[pattern];
 }
 
-engine.indexOf = function(coll, substr){
+engine.indexOf = function (coll, substr) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(substr) || util.isEmpty(str) ? [] : str.indexOf(substr);
 };
 
-engine.substring = function(coll, start, length){
+engine.substring = function (coll, start, length) {
   const str = misc.singleton(coll, 'String');
   if (util.isEmpty(str) || util.isEmpty(start) || start < 0 || start >= str.length) {
-    return  [];
+    return [];
   }
   if (length === undefined || util.isEmpty(length)) {
     return str.substring(start);
   }
   return str.substring(start, start + length);
 };
 
-engine.startsWith = function(coll, prefix){
+engine.startsWith = function (coll, prefix) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(prefix) || util.isEmpty(str) ? [] : str.startsWith(prefix);
 };
 
-engine.endsWith = function(coll, postfix) {
+engine.endsWith = function (coll, postfix) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(postfix) || util.isEmpty(str) ? [] : str.endsWith(postfix);
 };
 
-engine.containsFn = function(coll, substr){
+engine.containsFn = function (coll, substr) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(substr) || util.isEmpty(str) ? [] : str.includes(substr);
 };
 
-engine.upper = function(coll){
+engine.upper = function (coll) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(str) ? [] : str.toUpperCase();
 };
 
-
-engine.lower = function(coll){
+engine.lower = function (coll) {
   const str = misc.singleton(coll, 'String');
   return util.isEmpty(str) ? [] : str.toLowerCase();
 };
 
+engine.joinFn = function (coll, separator) {
+  const stringValues = coll.map((n) => {
+    const d = util.valData(n);
+    if (typeof d === "string") {
```

Comment:
The case where the element of the input collection is not a string is not described here: https://github.com/HL7/FHIRPath/pull/10/files
Just wondering, maybe in this case we need to generate an error?
I think the spec should clearly state what to do with non-string values.